src/ngircd/irc-login.c

```diff
@@ -259,6 +260,21 @@ IRC_NICK( CLIENT *Client, REQUEST *Req )
 				Client_SetType( Client, CLIENT_GOTNICK );
 		} else {
 			/* Nickname change */
+
+			/* Check that the user isn't on any channels set +N */
+			if(!Client_HasMode(Client, 'o')) {
```

Comment:
Without any testing, I think you have to make sure that the `Client` ist a `CLIENT_USER` (and no `CLIENT_SERVER` or `CLIENT_SERVICE`) here. No?